LoopKit/HealthKitSampleStore.swift

```diff
@@ -214,16 +219,40 @@ extension HealthKitSampleStore: HKSampleQueryTestable {
 }
 
 
+// MARK: - Query
+extension HealthKitSampleStore {
+    internal func predicateForSamples(withStart startDate: Date?, end endDate: Date?, options: HKQueryOptions = []) -> NSPredicate? {
+        guard observeHealthKitSamplesFromCurrentApp || observeHealthKitSamplesFromOtherApps else {
+            return nil
+        }
+
+        var predicate = HKQuery.predicateForSamples(withStart: startDate, end: endDate, options: options)
+
+        if !observeHealthKitSamplesFromCurrentApp {
```

Comment:
Wow...I had to stare at this boolean logic a while to make sure it was all good.  Nice job!  

[nit] maybe a comment in here would help future readers?  Something like
```
// Exclude current app data, but include other app data
```
```
// Include current app data, exclude other app data
```
?

Reply:
I will put this in a subsequent PR.